hdl/projects/sidecar/mainboard/PCIeEndpointController.bsv

```diff
@@ -43,6 +46,22 @@ module mkPCIeEndpointController
     ConfigReg#(Bool) host_reset <- mkConfigReg(False);
     ConfigReg#(Bool) power_fault <- mkConfigReg(False);
 
+    // This represents the debounced PCIe PERST signal. This signal is driven by
+    // a buffer whose input comes from Gimlet. This signal will oscillate during
+    // Gimlet reboot and thus we should lightly debounce it. We've chosen to
+    // apply a 200us debounce to rising and falling edge. The default state will
```

Comment:
We arrived at 200us per @ericaasen's analysis on #202:
>During a normal boot-up sequence, the pulse width of PERST going high is almost 2ms, so a 200us debounce on it would be a good start.

and

>I checked the Gen3 PCIe specs as well as the Tofino datasheet and design guide and did not see any requirements for minimum pulse widths for PERST# or PCIE_RST_N (Tofino name), so actual debounce time is a judgement call, but should be >0 and <1.5ms and be symmetrical (same on both rising and falling transitions) such that a valid reset signal is just time-shifted by the debounce and does not have its width modified.

Reply:
Given the PCIe spec says that `Tperst` has a minimum value of 100us, perhaps that should be our debounce limit? Otherwise we could technically follow spec by asserting PERST# for 100us but we'd lose that pulse with a 200us debounce.

Reply:
So there are a combination of things here to think about. The way that this works from the SP3 side is that it is pulled down and then when we get to A0, the PCA9535 that drives the resets wakes up. However, in @ericaasen's trace we see a brief blip which is probably the SMU changing things from an input to output maybe without properly changing the polarity first. This is because the PCA9535 reset value is to drive a value high. `Tperst-clk` is 100us and the various `Tpsert` values are 100us.

In practice, PERST is only ever really changed at slow ratios. That is, I don't think we'd ever see a strobe that was that short based on our system design, meaning 200us should be safe. But I think that if we want to play by the spec we should probably go with 100us. If we apply that filter to that would that get rid of the noise while we're draining?

Reply:
I understand that 200us is fine in practice, but I do want to play by the spec. 100us looks perfectly fine given the scope shots in #202. I'll update the PR to reflect the shorter debounce.